device/src/mac.rs

```diff
@@ -6,20 +6,24 @@ During Uplink assembly, this struct will be inquired to drive construction
 use heapless::Vec;
 
 use super::region;
-use lorawan_encoding::maccommands::{LinkADRAnsPayload, MacCommand};
+use lorawan_encoding::maccommands::{
+    LinkADRAnsPayload, MacCommand, RXTimingSetupAnsPayload, RXTimingSetupReqPayload,
+};
 
 #[derive(Default, Debug)]
 pub struct Mac {
     adr_ans: AdrAns,
+    rx_delay_ans: RxDelayAns,
 }
 
 type AdrAns = u8;
+type RxDelayAns = bool;
 
 //work around for E0390
 trait AdrAnsTrait {
     fn add(&mut self);
     fn clear(&mut self);
-    fn get(&mut self) -> u8;
+    fn get(&self) -> u8;
```

Comment:
nit: could you add a short comment maybe on the type level what it's used for? It might make reading the code slightly easier next time.